walletnode/services/senseregister/service.go

```diff
@@ -0,0 +1,85 @@
+package senseregister
+
+import (
+	"context"
+
+	"github.com/pastelnetwork/gonode/common/errors"
+	"github.com/pastelnetwork/gonode/common/service/artwork"
+	"github.com/pastelnetwork/gonode/common/service/task"
+	"github.com/pastelnetwork/gonode/common/storage"
+	"github.com/pastelnetwork/gonode/pastel"
+	"github.com/pastelnetwork/gonode/walletnode/node"
+)
+
+// Service represents a service for the registration artwork.
+type Service struct {
+	*task.Worker
+	*artwork.Storage
+
+	config       *Config
+	pastelClient pastel.Client
+	nodeClient   node.Client
+	db           storage.KeyValue
+}
+
+// Run starts worker.
+func (service Service) Run(ctx context.Context) error {
+	// TODO: implement service run logic
+
+	return nil
+}
+
+// AddTask adds a task to the worker.
+func (service *Service) AddTask(ticket *Request) string {
+	task := NewTask(service, ticket)
+	service.Worker.AddTask(task)
+
+	return task.ID()
+}
+
+// VerifyImageSignature verifies the signature of the image
+func (service *Service) VerifyImageSignature(ctx context.Context, file *artwork.File, signature string, pastelID string) error {
+	imgData, err := file.Bytes()
```

Comment:
should verify if file is nil

Reply:
File is openned before that, I will improve to re-used imgdata instead.